context/UserInactivity.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from "react";
+import { AppState, AppStateStatus, NativeEventSubscription } from 'react-native';
+import { secureStorage } from "~/lib/secureStorage";
+import { INACTIVITY_THRESHOLD } from "~/lib/constants";
+import { useAppStore } from "~/lib/state/appStore";
+
+export const UserInactivityProvider = ({ children }: any) => {
+  const [appState, setAppState] = React.useState<AppStateStatus>(AppState.currentState);
+  const isSecurityEnabled = useAppStore((store) => store.isSecurityEnabled);
+
+  const handleAppStateChange = async (nextState: AppStateStatus) => {
+    if (appState === "active" && nextState.match(/inactive|background/)) {
```

Comment:
Should we use `lastActiveTimeKey` here? 

https://github.com/getAlby/go/pull/120/files#diff-11ddfe4eefed7fd41505dfa00d0993f731137870defccdbfea779a76770d4bbaR38